### PR TITLE
Filter out controller pod statistics from service mesh metric requests

### LINKED ISSUE #370: Control pod stats appear in web UI when proxy is injected into control pod
After running install with the newly added functionality which injects an "mesh" proxy into the control pod. We see telemetry stats for the "mesh" proxy in the web UI. At first glance, this may seem confusing to a user when they install their services in a separate namespace. One solution is to either disable mesh telemetry stats in the short term and revisit how we would like for that information to show up in the UI. 

__Using emojivoto as a test environment, we see that `conduit/controller` shows up in the UI__

screen shot 2018-02-16 at 10 05 37 am

__Clicking on the `emojivoto/web` deployment, we see `conduit/controller` stats show up again__

screen shot 2018-02-16 at 10 06 20 am

__However, when we click on the link to view the `conduit/controller` deployment stats, we get an error__

failed-controller-stats
 
/cc @briansmith

### PR DESCRIPTION
**Pull request by dadjeibaah**
When the conduit proxy is injected into the controller pod, we observe controller pod proxy stats show up as an "outbound" deployment for an unrelated upstream deployment. This may cause confusion when monitoring deployments in the service mesh. 

This PR filters out this "misleading" stat in the public api whenever the dashboard requests metric information for a specific deployment.

fixes #370

Signed-off-by: Dennis Adjei-Baah <dennis@buoyant.io>

### CODE REVIEW

**Comment by briansmith** [comment]
I took myself out of the reviewer list for this. I'm not a good reviewer for this.

**Comment by dadjeibaah** [comment]
Now that I think of it, it does. Do we have any metric label that distinguish metrics that are specific to a `controller`, `prometheus` and `web` deployments? I could make the exclusion rule more specific but I feel that might be a little too much for prometheus query.

**Comment by adleong** [comment]
I don't think there's any metric label other than checking the names against "controller", "prometheus", and "web".

For what it's worth, I still think it would be easier to simply turn off reporting in the controller sidecars.

**Comment by dadjeibaah** [comment]
Tried the `CONDUIT_PROXY_EVENT_BUFFER_CAPACITY` and its definitely a cleaner approach. Thanks for the info @adleong

File: cli/cmd/install.go [hunk 1]
```diff
@@ -85,7 +85,17 @@ func render(config installConfig, w io.Writer) error {
...
-	return InjectYAML(buf, w, conduitVersion)
+
+	proxyConfig := &proxyConfig{
+		version:         conduitVersion,
+		logLevel:        proxyLogLevel,
+		apiPort:         proxyAPIPort,
+		controlPort:     proxyControlPort,
+		outboundPort:    outboundPort,
+		inboundPort:     inboundPort,
+		eventBufferSize: 0,
```
Comment (adleong) [inline cli/cmd/install.go]: Perhaps add a comment here explaining that this is for turning off telemetry reporting from the controller proxies.

**Comment by olix0r** [comment]
I think it would be really really valuable to get this stuff into prometheus, but it's really not worth bikeshedding this as it all has to change after this release

This change _does_ strike me as quite a bit larger than i expected -- i honestly would have been fine just dropping all metrics in the namespace because it's simple and temporary.

**Comment by olix0r** [comment]
I suppose my main issue with this is that we're not going to want to have a configurable event buffer size for more than the 0.3.1 release -- so all of this work will quickly become moot.

**Comment by dadjeibaah** [comment]
Closing this PR in favor of #493

**Pull request closed without merging**